geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/utils/ConfluentMetadata.scala

```diff
@@ -0,0 +1,129 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.kafka.utils
+
+import java.util.Date
+import java.util.concurrent.TimeUnit
+
+import com.github.benmanes.caffeine.cache.{CacheLoader, Caffeine, LoadingCache}
+import com.typesafe.scalalogging.LazyLogging
+import org.locationtech.jts.geom.Geometry
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient
+import org.apache.avro.Schema
+import org.apache.avro.Schema.Type._
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder
+import org.locationtech.geomesa.features.confluent.ConfluentFeatureSerializer
+import org.locationtech.geomesa.index.metadata.GeoMesaMetadata
+import org.locationtech.geomesa.kafka.data.KafkaDataStore
+import org.locationtech.geomesa.kafka.utils.ConfluentMetadata.{schemaToSft, SUBJECT_POSTFIX}
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.opengis.feature.simple.SimpleFeatureType
+
+import scala.collection.JavaConverters._
+import scala.util.control.NonFatal
+
+class ConfluentMetadata(val schemaRegistry: SchemaRegistryClient) extends GeoMesaMetadata[String] with LazyLogging {
+
+  val topicSftCache: LoadingCache[String, String] =
```

Comment:
could we make this private? generally it makes things simpler if you don't expose things that aren't needed externally.